### ISSUE TITLE
Serverless deploy does not pick up env vars from .env.production, when also defined in .env.defaults

### ISSUE THREAD

**Issue by dac09** [redwoodjs/graphql#4548]
Say you have a `.env.defaults` with `DATABASE_URL` defined

```
#.env.defaults file
DATABASE_URL=postgresql://postgres@localhost:5432/deploy-target-serverless
```

And after running `yarn rw deploy sls --first-run`, and adding your DATABASE_URL varyour `.env.production` looks like this:

```
#.env.production
DATABASE_URL=postgresql://postgres:xxxxx@mydbprovider.whatever.com/postgres?connection_limit=1
API_URL=xxx
```

The deploys graphql lambda will use the DB defined in the defaults file.

### Expected behaviour 
Env vars should be loaded like:

.env.defaults -> .env -> .env.production, each one overriding anything with the same name

**Comment by dac09** [comment]
Yes @thedavidprice I did confirm this behaviour! I just used your project, but removed the org/app, and since I didn't have your local `.serverless` cached files or `.env.production` I think was a legit repro 👍

**Comment by thedavidprice** [comment]
## Update
Confirming this is still a problem, which is exacerbated because we instruct users to create `.env.production` in the Serverless Deploy Docs. Based on the docs, it's expected 1) you should continue to use .env.production and 2) redwood supports .env.context file naming. It does not.

See also https://github.com/redwoodjs/redwood/issues/5785 for an example of the confusion this creates.

## Next Steps
We need to determine if we want to support this behavior. At this time, it's a low priority. If someone wants to take it on, we could support the effort.

### Update Documentation
A a minimum, we can update the Serverless Deploy docs and remove the `.env.production` instructions.

cc @jtoar @Josh-Walker-GM

**Comment by Yuvraj102** [comment]
Hello, I'd love to progress on this, Pls assign it to me.
